### PR TITLE
add new command zident for user operations in Auth0

### PR DESCRIPTION
**Pull request by alfred-landrum**
This adds a new command, zident, for user and tenant administration.
This initial checkin supports user operations, including creating a new
user, triggering a password reset for a user, and searching for users
with an email, user id, or tenant id.

### CODE REVIEW

File: ppl/cmd/zident/user/new.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		}
+		tenantID = c.tenantID
+	} else {
+		tenantID = newTenantID()
+	}
+
+	userID := newUserID()
+
+	m, err := management.New(c.a0cfg.domain.String(),
+		management.WithClientCredentials(c.a0cfg.clientId, c.a0cfg.clientSecret))
```
Comment (marktwallace) [inline ppl/cmd/zident/user/new.go:63]: This is our vendor secret from Auth0, right? How does c.a0cfg, which contains the secrets, become available to this command?

File: ppl/cmd/zident/user/user.go [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+		err = multierr.Append(err, errors.New("AUTH0_CLIENT_ID not set"))
+	}
+	if c.clientSecret = os.Getenv("AUTH0_CLIENT_SECRET"); c.clientSecret == "" {
+		err = multierr.Append(err, errors.New("AUTH0_CLIENT_SECRET not set"))
+	}
+	if c.connection = os.Getenv("AUTH0_CONNECTION"); c.connection == "" {
+		err = multierr.Append(err, errors.New("AUTH0_CONNECTION not set"))
+	}
+	domain := os.Getenv("AUTH0_DOMAIN")
+	if domain == "" {
```
Comment (marktwallace) [inline ppl/cmd/zident/user/user.go:43]: I assume that Auth0 recommends the use of these env vars, right? So we will include the env vars in the deployment.yaml with references to secrets. This is all good and expected, but now our distributed configuration will include a mix of env vars and command line flags.

File: ppl/cmd/zident/user/user.go [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+	}); err != nil {
+		return nil, err
+	}
+	if user == nil {
+		return nil, errors.New("no user found")
+	}
+	return user, nil
+}
+
+func streamUsers(ctx context.Context, m *management.Management, f searchFlags, allowMultiple bool, fn func(*management.User) error) error {
```
Comment (marktwallace) [inline ppl/cmd/zident/user/user.go:85]: "management" is kind of a vague name for the main Auth0 interface. Now that I see it, I will expect it in this context, so NP.

**Review by marktwallace — APPROVED** [review]
This all looks straightforward and right. One piece of context I am missing is whether or not Auth0 has CLI tools that do similar things. This functionality is so generic that I would not be surprised if Auth0 has CLI in addition to the Golang API you are using for these basic user management operations.

**Review by mattnibs — APPROVED** [review]
I imagine at some point we'll want to move some of this functionality (e.g. the user lookup methods) into its own package that can be accessed by both this command and zqd alike?

**Pull request closed without merging**